tests/parser/types/numbers/test_sqrt.py

```diff
@@ -0,0 +1,161 @@
+from decimal import (
+    ROUND_FLOOR,
+    Decimal,
+    getcontext,
+)
+
+from eth_tester.exceptions import (
+    TransactionFailed,
+)
+import hypothesis
+import pytest
+
+from vyper.utils import (
+    SizeLimits,
+)
+
+getcontext().prec = 168
+DECIMAL_PLACES = 10
+DECIMAL_RANGE = [
+    Decimal('0.' + '0' * d + '2')
+    for d in range(0, DECIMAL_PLACES)
+]
+
+
+def decimal_truncate(val, decimal_places=DECIMAL_PLACES, rounding=ROUND_FLOOR):
+    q = '0'
+    if decimal_places != 0:
+        q += '.' + '0' * decimal_places
+
+    return val.quantize(Decimal(q), rounding=rounding)
+
+
+def decimal_sqrt(val):
+    return decimal_truncate(val.sqrt())
+
+
+def test_sqrt_literal(get_contract_with_gas_estimation):
+    code = """
+@public
+def test() -> decimal:
+    return sqrt(2.0)
+    """
+    c = get_contract_with_gas_estimation(code)
+    assert c.test() == decimal_sqrt(Decimal('2'))
+
+
+def test_sqrt_variable(get_contract_with_gas_estimation):
+    code = """
+@public
+def test(a: decimal) -> decimal:
+    return sqrt(a)
+
+@public
+def test2() -> decimal:
+    a: decimal = 44.001
+    return sqrt(a)
+    """
+
+    c = get_contract_with_gas_estimation(code)
+
+    val = Decimal('33.33')
+    assert c.test(val) == decimal_sqrt(val)
+
+    val = Decimal('0.1')
+    assert c.test(val) == decimal_sqrt(val)
+
+    assert c.test(Decimal('0.0')) == Decimal('0.0')
+    assert c.test2() == decimal_sqrt(Decimal('44.001'))
+
+    with pytest.raises(TransactionFailed):
+        c.test(Decimal('-1.1'))
+
```

Comment:
Also do this to show the "smallest" possible negative number reverts (furthest away from zero we can test for)
```suggestion

    with pytest.raises(TransactionFailed):
        c.test(Decimal(SizeLimits.MINNUM))
```

Reply:
Can we also test max decimal (binary representation `0xff..`)

Reply:
P.S. there's a way to cherry pick specific cases with hypothesis

Reply:
@fubuloubu is referring to https://hypothesis.readthedocs.io/en/latest/reproducing.html#hypothesis.example

Reply:
@fubuloubu I added and additional test for the invalid range.

Reply:
Perfect!